File: jsimple-oauth/c#/jsimple/oauth/utils/MapUtils.cs
using System.Collections.Generic;
using System.Text;

namespace jsimple.oauth.utils
{


	/// <summary>
	/// @author: Pablo Fernandez
	/// </summary>
	public class MapUtils
	{
		public static string ToString<K, V>(IDictionary<K, V> map)
		{
			if (map == null)
				return "";
			if (map.Count == 0)
				return "{}";

			StringBuilder result = new StringBuilder();
			result.Append("{");

			bool addedSomething = false;
			foreach (KeyValuePair<K, V> entry in map)
			{
				if (addedSomething)
					result.Append(", ");
				else
					addedSomething = true;

				result.Append(entry.Key.ToString());
				result.Append(" -> ");
				result.Append(entry.Value.ToString());

				result.Append(string.Format(", {0} -> {1} ", entry.Key.ToString(), entry.Value.ToString()));
			}

			result.Append("}");

			return result.ToString();
		}
	}

}

File: jsimple-oauth/src/main/java/jsimple/oauth/utils/MapUtils.java
package jsimple.oauth.utils;

import java.util.Map;

/**
 * @author: Pablo Fernandez
 */
public class MapUtils {
    public static <K, V> String toStringFromMap(Map<K, V> map) {
        if (map == null) return "";
        if (map.isEmpty()) return "{}";

        StringBuilder result = new StringBuilder();
        result.append("{");

        boolean addedSomething = false;
        for (Map.Entry<K, V> entry : map.entrySet()) {
            if (addedSomething)
                result.append(", ");
            else addedSomething = true;

            result.append(entry.getKey().toString());
            result.append(" -> ");
            result.append(entry.getValue().toString());

            result.append(String.format(", %s -> %s ", entry.getKey().toString(), entry.getValue().toString()));
        }

        result.append("}");

        return result.toString();
    }
}
